### PR TITLE
support coredns v1.11.1

### PR DESCRIPTION
**Pull request by tukwila**
Now coredns release new version:
https://github.com/coredns/coredns/releases/tag/v1.11.1
so do upgrade in corefile-migration.

```
docker.io/coredns/coredns:1.11.1:                                                 resolved       |++++++++++++++++++++++++++++++++++++++| 
index-sha256:1eeb4c7316bacb1d4c8ead65571cd92dd21e27359f0d4917f1a5822a73b75db1:    done           |++++++++++++++++++++++++++++++++++++++| 
manifest-sha256:2169b3b96af988cf69d7dd69efbcc59433eb027320eb185c6110e0850b997870: done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:c54d895c1975f6949e8b6d1457478f749f801f4622afa1336081b95b0bda5641:    done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:02cd68c0cbf64abe9738767877756b33f50fff5d88583fdc74b66beffa77694b:    done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:d3c894b5b2b0fa857549aeb6cbc38b038b5b2828736be37b6d9fff0b886f12fd:    done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:b40161cd83fc5d470d6abe50e87aa288481b6b89137012881d74187cfbf9f502:    done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:46ba3f23f1d3fb1440deeb279716e4377e79e61736ec2227270349b9618a0fdd:    done           |++++++++++++++++++++++++++++++++++++++| 
config-sha256:cbb01a7bd410dc08ba382018ab909a674fb0e48687f0c00797ed5bc34fcc6bb4:   done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:4fa131a1b726b2d6468d461e7d8867a2157d5671f712461d8abd126155fdf9ce:    done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:dd5ad9c9c29f04b41a0155c720cf5ccab28ef6d353f1fe17a06c579c70054f0a:    done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:860aeecad37138b467a6c41d5278c647ed77350db18b56364c29a4770e271fca:    done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:b4ca4c215f483111b64ec6919f1659ff475d7080a649d6acd78a6ade562a4a63:    done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:eebb06941f3e57b2e40a0e9cbd798dacef9b04d89ebaa8896be5f17c976f8666:    done           |++++++++++++++++++++++++++++++++++++++| 
layer-sha256:960043b8858c3c30f1d79dcc49adb2804fd35c2510729e67685b298b2ca746b7:    done           |++++++++++++++++++++++++++++++++++++++| 
elapsed: 24.2s                                                                    total:  17.1 M (724.2 KiB/s)                                     
unpacking linux/amd64 sha256:1eeb4c7316bacb1d4c8ead65571cd92dd21e27359f0d4917f1a5822a73b75db1...
done: 10.175185987s     

```

It seems that there is no special change or plugin update from v1.11.0 to v1.11.1 that may influence corefile migration.

### CODE REVIEW

**Comment by tukwila** [comment]
@chrisohaver please review PR, big thks. -:)

File: corefile-tool/cmd/validversions_test.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ func TestNewValidVersionsCmd(t *testing.T) {
 		{
 			name: "Works without error",
 			expectedOutput: `The following are valid CoreDNS versions:
-1.1.3, 1.1.4, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.2.4, 1.2.5, 1.2.6, 1.3.0, 1.3.1, 1.4.0, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.6.1, 1.6.2, 1.6.3, 1.6.4, 1.6.5, 1.6.6, 1.6.7, 1.6.9, 1.7.0, 1.7.1, 1.8.0, 1.8.3, 1.8.4, 1.8.5, 1.8.6, 1.8.7, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.9.4, 1.10.0, 1.10.1, 1.11.0
+1.1.3, 1.1.4, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.2.4, 1.2.5, 1.2.6, 1.3.0, 1.3.1, 1.4.0, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.6.1, 1.6.2, 1.6.3, 1.6.4, 1.6.5, 1.6.6, 1.6.7, 1.6.9, 1.7.0, 1.7.1, 1.8.0, 1.8.3, 1.8.4, 1.8.5, 1.8.6, 1.8.7, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.9.4, 1.10.0, 1.10.1, 1.11.0, 1.11.1
```
Comment (chrisohaver) [inline corefile-tool/cmd/validversions_test.go:18]: I think trailing space after version string is causing test to fail.

**Review by chrisohaver — CHANGES_REQUESTED** [review]

File: corefile-tool/cmd/validversions_test.go (hunk 1 above)
Comment (tukwila) [reply corefile-tool/cmd/validversions_test.go]: yes, i corrected.

**Comment by pacoxu** [comment]
A new release is needed for https://github.com/kubernetes/kubernetes/pull/120116#issuecomment-1692964735

**Comment by chrisohaver** [comment]
https://github.com/coredns/corefile-migration/releases/tag/v1.0.21

**Pull request merged**